bot.go

```diff
@@ -34,19 +34,36 @@ type BotAPI struct {
 //
 // It requires a token, provided by @BotFather on Telegram.
 func NewBotAPI(token string) (*BotAPI, error) {
-	return NewBotAPIWithClient(token, &http.Client{})
+	return CreateNewBotApi(token, &http.Client{}, false)
 }
 
 // NewBotAPIWithClient creates a new BotAPI instance
 // and allows you to pass a http.Client.
 //
 // It requires a token, provided by @BotFather on Telegram.
 func NewBotAPIWithClient(token string, client *http.Client) (*BotAPI, error) {
+	return CreateNewBotApi(token, client, false)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
+// and enables debug by default.
+//
+// It requires a token, provided by @BotFather on Telegram.
+func NewBotApiWithDebug(token string) (*BotAPI, error) {
+	return CreateNewBotApi(token, &http.Client{}, true)
+}
+
+// NewBotAPIWithClient creates a new BotAPI instance
+// and allows you to pass all additional customization that you might need.
+//
+// It requires a token, provided by @BotFather on Telegram.
+func CreateNewBotApi(token string, client *http.Client, debug bool) (*BotAPI, error) {
```

Comment:
Same here for the method name.

```suggestion
func CreateNewBotAPI(token string, client *http.Client, debug bool) (*BotAPI, error) {
```